reana_job_controller/app.py

```diff
@@ -55,212 +59,195 @@ def filter_jobs(job_db):
-                  "cmd": "sleep 1000",
-                  "cvmfs_mounts": [
-                    "atlas-condb",
-                    "atlas"
-                  ],
-                  "docker-img": "busybox",
-                  "experiment": "atlas",
-                  "job-id": "2e4bbc1d-db5e-4ee0-9701-6e2b1ba55c20",
-                  "max_restart_count": 3,
-                  "restart_count": 0,
-                  "status": "started"
+    ---
+    get:
+      description: Get all Jobs
+      produces:
+       - application/json
+      responses:
+        200:
+          description: Job list.
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/Job'
+          examples:
+            application/json:
+              {
+                "jobs": {
+                  "1612a779-f3fa-4344-8819-3d12fa9b9d90": {
+                    "cmd": "sleep 1000",
+                    "cvmfs_mounts": [
+                      "atlas-condb",
+                      "atlas"
+                    ],
+                    "docker_img": "busybox",
+                    "experiment": "atlas",
+                    "job_id": "1612a779-f3fa-4344-8819-3d12fa9b9d90",
+                    "max_restart_count": 3,
+                    "restart_count": 0,
+                    "status": "succeeded"
+                  },
+                  "2e4bbc1d-db5e-4ee0-9701-6e2b1ba55c20": {
+                    "cmd": "sleep 1000",
+                    "cvmfs_mounts": [
+                      "atlas-condb",
+                      "atlas"
+                    ],
+                    "docker_img": "busybox",
+                    "experiment": "atlas",
+                    "job_id": "2e4bbc1d-db5e-4ee0-9701-6e2b1ba55c20",
+                    "max_restart_count": 3,
+                    "restart_count": 0,
+                    "status": "started"
+                  }
                 }
               }
-            }
-
-        :resheader Content-Type: application/json
-        :statuscode 200: no error - the list has been returned.
     """
+    # FIXME do Marshmallow validation after fixing the structure
+    # of job list. Now it has the ID as key, it should be a plain
+    # list of jobs so it can be validated with Marshmallow.
+
     return jsonify({"jobs": filter_jobs(JOB_DB)}), 200
 
 
 @app.route('/jobs', methods=['POST'])
-def create_job():
+def create_job():  # noqa
     """Create a new job.
 
-    .. http:post:: /jobs
-
+    ---
+    post:
+      summary: |-
         This resource is expecting JSON data with all the necessary
         information of a new job.
+      description: Create a new Job.
+      operationId: create_job
+      consumes:
+       - application/json
+      produces:
+       - application/json
+      parameters:
+       - name: job
+         in: body
+         description: Information needed to instantiate a Job
+         required: true
+         schema:
+           $ref: '#/definitions/JobRequest'
+      responses:
```

Comment:
- 201: Request succeeded. The job has been launched.
- 400: Request failed. The incoming data specification seems malformed.
- 500: Request failed. Internal controller error. The job could probably not have been allocated.